src/poll.rs

```diff
@@ -475,7 +475,9 @@ mod syscall {
         let fd = unsafe { hermit_abi::eventfd(count, 0) };
 
         if fd == -1 {
```

Comment:
` if fd == -1 {`

You should use ` if fd < 0 {`. In our case, the return values could be another negative number.